dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -429,17 +429,81 @@ endif::[]
 [[using-nodeport]]
 == Using a Service NodePort
 
-Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster.
-NodePorts are in the 30-60k range by default, which means a NodePort is
-unlikely to match a service's intended port (for example, 8080 may be exposed
-as 31020). This use of ports is wasteful of scarce port resources.
-However, it is slightly easier to set up. Again, this requires more privileges.
+Use xref:../architecture/core_concepts/pods_and_services.adoc#service-nodeport[NodePorts] to expose the service nodePort on all nodes in the cluster. NodePorts are intended to allow load balancing across an entire cluster and are used by the LoadBalancer service type when running on a cloud provider. You can specify a specific nodePort (if not already in use) or allow the system to assign you one. When the nodePort is assigned, all nodes in the cluster will allow incoming traffic to that port and then direct the traffic from that port to a local or remote endpoint under the service. By default, {product-title} allocates ports between `30000` and `32000` as NodePorts. For very high density environments, you may need to increase the range of ports. OpenShift automatically opens the node port range at install time for firewalls and cloud security groups for new clusters.
```

Comment:
s/NodePorts are intended to allow/NodePorts allow
port, and then direct
s/OpenShift/{product-title}
s/node port range/nodePort range
s/at install time/during installation